tables/vpc_flow_log/vpc_flow_log_extractor.go

```diff
@@ -82,148 +100,15 @@ func ConvertToMapSlice(data string) ([]map[string]string, error) {
-			}
-			v := int32(protocol)
-			flowLog.Protocol = &v
-		case "packets":
-			packets, err := strconv.ParseInt(field, 10, 64)
-			if err != nil {
-				return fmt.Errorf("invalid packets: %s", field)
-			}
-			flowLog.Packets = &packets
-		case "bytes":
-			bytes, err := strconv.ParseInt(field, 10, 64)
-			if err != nil {
-				return fmt.Errorf("invalid bytes: %s", field)
-			}
-			flowLog.Bytes = &bytes
-		case "start":
-			start, err := strconv.ParseInt(field, 10, 64)
-			if err != nil {
-				return fmt.Errorf("invalid start value: %s", field)
-			}
-			t := time.Unix(start, 0)
-			flowLog.Start = &t
-		case "end":
-			end, err := strconv.ParseInt(field, 10, 64)
-			if err != nil {
-				return fmt.Errorf("invalid end value: %s", field)
-			}
-			t := time.Unix(end, 0)
-			flowLog.End = &t
-		case "action":
-			flowLog.Action = &field
-		case "log-status":
-			flowLog.LogStatus = &field
-		case "vpc-id":
-			flowLog.VPCID = &field
-		case "subnet-id":
-			flowLog.SubnetID = &field
-		case "instance-id":
-			flowLog.InstanceID = &field
-		case "tcp-flags":
-			tcpFlags, err := strconv.ParseInt(field, 10, 32)
-			if err != nil {
-				return fmt.Errorf("invalid tcp-flags: %s", field)
-			}
-			v := int32(tcpFlags)
-			flowLog.TCPFlags = &v
-		case "type":
-			flowLog.Type = &field
-		case "pkt-srcaddr":
-			flowLog.PktSrcAddr = &field
-		case "pkt-dstaddr":
-			flowLog.PktDstAddr = &field
-		case "region":
-			flowLog.Region = &field
-		case "reject-reason":
-			flowLog.RejectReason = &field
-		case "az-id":
-			flowLog.AzID = &field
-		case "sublocation-type":
-			flowLog.SublocationType = &field
-		case "sublocation-id":
-			flowLog.SublocationID = &field
-		case "pkt-src-aws-service":
-			flowLog.PktSrcAWSService = &field
-		case "pkt-dst-aws-service":
-			flowLog.PktDstAWSService = &field
-		case "flow-direction":
-			flowLog.FlowDirection = &field
-		case "traffic-path":
-			trafficPath, err := strconv.ParseInt(field, 10, 32)
-			if err != nil {
-				return fmt.Errorf("invalid traffic-path: %s", field)
-			}
-			v := int32(trafficPath)
-			flowLog.TrafficPath = &v
-		case "ecs-cluster-arn":
-			flowLog.ECSClusterARN = &field
-		case "ecs-cluster-name":
-			flowLog.ECSClusterName = &field
-		case "ecs-container-instance-arn":
-			flowLog.ECSContainerInstanceARN = &field
-		case "ecs-container-instance-id":
-			flowLog.ECSContainerInstanceID = &field
-		case "ecs-container-id":
-			flowLog.ECSContainerID = &field
-		case "ecs-second-container-id":
-			flowLog.ECSSecondContainerID = &field
-		case "ecs-service-name":
-			flowLog.ECSServiceName = &field
-		case "ecs-task-definition-arn":
-			flowLog.ECSTaskDefinitionARN = &field
-		case "ecs-task-arn":
-			flowLog.ECSTaskARN = &field
-		case "ecs-task-id":
-			flowLog.ECSTaskID = &field
-		default:
-			return fmt.Errorf("unknown field: %s", i)
+	for _, field := range header {
+		if !slices.Contains(allLogFields, field) {
+			headerAvailable = false
+			break
 		}
```

Comment:
As mentioned above rather than using `allLogFields` can check if field is a valid key in the map from getValidTokensAndColumnNames()

Reply:
Yes, now the check had been added using the `getValidTokensAndColumnNames()`.